src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -3335,21 +3334,23 @@ module EstablishTypeDefinitionCores =
 
                       | [(ty, m)] ->
                           let inheritRange =
-                                match inherits with
-                                | [] -> m
-                                | (synType, _, _) :: _ -> synType.Range
+                              match inherits with
+                              | [] -> m
+                              | (synType, _, _) :: _ -> synType.Range       
                           if not firstPass && not (match kind with SynTypeDefnKind.Class -> true | _ -> false) then
                               errorR (Error(FSComp.SR.tcStructsInterfacesEnumsDelegatesMayNotInheritFromOtherTypes(), inheritRange)) 
                           CheckSuperType cenv ty inheritRange 
                           if isTyparTy g ty then 
                               if firstPass then 
                                   errorR(Error(FSComp.SR.tcCannotInheritFromVariableType(), inheritRange)) 
                               Some g.obj_ty_noNulls // a "super" that is a variable type causes grief later
-                          else             
-                                       
+                          else                  
                               Some ty 
-                      | _ -> 
-                          error(Error(FSComp.SR.tcTypesCannotInheritFromMultipleConcreteTypes(), m))
+                      | _ ->
+                          inherits
+                          |> List.skip 1
```

Comment:
Please lean on the compiler to make sure the list length will be valid, instead of the universal `_` clause together with potentially unsafe `List.skip` (it is not unsafe now, but leaning on the pattern match completeness will be more robust against future refactorings)

Like ` | _first :: superFluousInherits `

Reply:
Sure.